share/picongpu/tests/CurrentDeposition/lib/python/test/CurrentDeposition/AfuncW.py

```diff
@@ -0,0 +1,72 @@
+"""
+This file is part of the PIConGPU.
+
+Copyright 2023 PIConGPU contributors
+Authors: Hannes Wolf
+License: GPLv3+
+
+implements the assignment functions of order 0 to 3 as well as a function to
+calculate the current deposition vector
+"""
+
+
+# functions for single values
+def NGP(x):
+    """generates hat function with width 1 (symmetrical around x=0) -
+    assignement function of Order 0
+    """
+
+    if abs(x) < 1/2:
+        y = 1
+    elif abs(x) == 1/2:
+        y = 1/2
+    else:
+        y = 0
+    return y
+
+
+def CIC(x):
+    """generates hat triangle with width 2 (symmetrical around x=0) -
+    assignement function of order 1 for single values
+    """
+
+    if abs(x) < 1:
+        y = 1 - abs(x)
+    else:
+        y = 0
+    return y
+
+
+def TSC(x):
+    """assignementfunction of order 2" for single values"""
+
+    if abs(x) < 3/2 and abs(x) >= 1/2:
+        y = ((3/2 - abs(x))**2)/2
+    elif abs(x) < 1/2:
+        y = (-(x**2) + 3/4)
+    else:
+        y = 0
+    return y
+
+
+def PQS(x):
+    """assignmentfunction of order 3 for single values"""
+
+    if abs(x) < 2 and abs(x) >= 1:
+        y = 1/6*(2 - abs(x))**3
+    elif abs(x) < 1:
+        y = (2/3 - x**2 + (abs(x)**3)/2)
+    else:
+        y = 0
+    return y
+
+
+def W(s1, s2, s3, s4, s5, s6):
+    """Calculation of the current deposition; s1, s2, ..., s6
+    are the values of the assignement function at the old and the new
+    coordinates
+    s1, s2, ..., s6 = assignment-function(i - cell_position)"""
+
+    W = 1/3*(s4*s5*s6 - s1*s5*s6 + s4*s2*s3 - s1*s2*s3) + (
+        1/6*(s4*s2*s6 - s1*s2*s6 + s4*s5*s3 - s1*s5*s3))

```

Comment:
There is superfluous parenthesis around the second term. Either use `\` at the end of the first line or put the parenthesis around the whole expression.

Reply:
done